R/use_lintr.R

```diff
@@ -43,5 +43,21 @@ use_lintr <- function(path = ".", type = c("tidyverse", "full")) {
     )
   )
   write.dcf(the_config, config_file, width = Inf)
+
+  # Check if config_file is in package i.e. lintr_option("linter_file") != "../.lintr"
+  pkg_path <- normalizePath(path, mustWork = FALSE, winslash = "/")
+  if(startsWith(config_file, prefix = pkg_path)) {
+    # Skip a extra character for the leading `/`
+    rel_path <- substring(config_file, first = nchar(pkg_path) + 2L, last = nchar(config_file))
```

Comment:
Not 100% sure but I remember package tools has some helpers for these operations.

Reply:
The only thing I can find is this `tools::file_path_as_absolute(".")`, but it's just a wrapper around `normalizePath`. If you know something specific, I'd be happy to include it.

Reply:
NVM I misremembered. They are in {fs} https://search.r-project.org/CRAN/refmans/fs/html/path_math.html

Since we don't import it, let's stick to base R. Thanks for checking.